src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -21,63 +55,440 @@ const ActorInit En_Dns_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFAFC.s")
+DnsItemEntry D_809F0470 = { 10, 1, GI_HEART_PIECE, func_809EF70C, func_809EFA58 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFB40.s")
+DnsItemEntry D_809F0480 = { 40, 30, GI_SEEDS_30, func_809EF73C, func_809EF9F8 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFB84.s")
+DnsItemEntry D_809F0490 = { 50, 1, GI_SHIELD_DEKU, func_809EF800, func_809EF9F8 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFBC8.s")
+DnsItemEntry D_809F04A0 = { 40, 5, GI_BOMBS_5, func_809EF854, func_809EFA9C };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFC9C.s")
+DnsItemEntry D_809F04B0 = { 70, 20, GI_ARROWS_LARGE, func_809EF8F4, func_809EFACC };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFDD0.s")
+DnsItemEntry D_809F04C0 = { 40, 1, GI_POTION_RED, func_809EF9A4, func_809EF9F8 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFEE8.s")
+DnsItemEntry D_809F04D0 = { 40, 1, GI_POTION_GREEN, func_809EF9A4, func_809EF9F8 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFF50.s")
+DnsItemEntry D_809F04E0 = { 40, 1, GI_STICK_UPGRADE_20, func_809EF70C, func_809EFAFC };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809EFF98.s")
+DnsItemEntry D_809F04F0 = { 40, 1, GI_NUT_UPGRADE_30, func_809EF70C, func_809EFB40 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809F008C.s")
+DnsItemEntry* D_809F0500[] = {
+    &D_809F0450, &D_809F0460, &D_809F0470, &D_809F0480, &D_809F0490, &D_809F04A0,
+    &D_809F04B0, &D_809F04C0, &D_809F04D0, &D_809F04E0, &D_809F04F0,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809F0100.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_S8(naviEnemyId, 78, ICHAIN_CONTINUE),
+    ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 30, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/func_809F017C.s")
+typedef struct {
+    /* 0x00 */ AnimationHeader* anim;
+    /* 0x04 */ u8 mode;
+    /* 0x08 */ f32 transitionRate;
+} DnsAnimInfo; // size = 0xC
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/EnDns_Update.s")
+DnsAnimInfo D_809F0538[] = {
+    { 0x06001108, 0x00, 0.0f },
+    { 0x06004404, 0x02, 0.0f },
+    { 0x060009A0, 0x02, 0.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dns/EnDns_Draw.s")
+extern AnimationHeader D_060009A0;
+extern SkeletonHeader D_060041A8;
+extern GenericAnimationHeader D_06004404;
+
+void EnDns_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDns* this = THIS;
+
+    if (this->actor.params < 0) {
+        osSyncPrintf(VT_FGCOL(RED) "引数エラー（売りナッツ）[ arg_data = %d ]" VT_RST "\n", this->actor.params);
+        // FUNCTION ERROR (DEKU SALESMAN)
```

Comment:
We generally have translation above the print. Also avoid all caps when translating (we still have some of that in the codebase, from the old style, but they should get fixed at some point).
```suggestion
        // "Function Error (Deku Salesman)"
        osSyncPrintf(VT_FGCOL(RED) "引数エラー（売りナッツ）[ arg_data = %d ]" VT_RST "\n", this->actor.params);
```